Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/DepartmentServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package cz.fi.muni.pa165.secretagency.service;
+
+import cz.fi.muni.pa165.secretagency.dao.DepartmentDao;
+import cz.fi.muni.pa165.secretagency.entity.Department;
+import cz.fi.muni.pa165.secretagency.enums.DepartmentSpecialization;
+import cz.fi.muni.pa165.secretagency.service.exceptions.DepartmentServiceException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation of service access to the {@link Department} entity.
+ * This class is part of the service module of the application
+ * that provides the implementation of the business logic (main logic of the application).
+ *
+ * @author Milos Silhar (433614)
+ */
+@Service
+@Transactional
+public class DepartmentServiceImpl extends GenericServiceImpl<Department, DepartmentDao> implements DepartmentService {
+
+    @Override
+    public List<Department> getDepartmentByCity(String city) {
+        return getDao().getDepartmentsInCity(city);
```

Comment:
Přidal bych ošetření city na null.

Reply:
Len city može byť u department nastavené na null, tak som to tam nechal že ide vyhľadať aj tie department kde neni uvedené city.

Reply:
Tak v tom případě to je správně. :)